mesonbuild/backend/backends.py

```diff
@@ -1168,24 +1168,30 @@ def determine_windows_extra_paths(
         links to and return them so they can be used in unit
         tests.
         """
-        result: T.Set[str] = set()
-        prospectives: T.Set[build.BuildTargetTypes] = set()
+        result: mesonlib.OrderedSet[str] = mesonlib.OrderedSet()
+        prospectives: mesonlib.OrderedSet[build.BuildTargetTypes] = mesonlib.OrderedSet()
```

Comment:
I wonder if this could be a `list` instead, knowing that deduplication will be performed when putting back prospectives into result. My concern is about performance, knowing that `OrderedSet` are slower than `set`, and that this function is called many times during setup. I think you should profile to ensure things do not slow down...

Reply:
Hi @bruchar1! Do you mean a `set`, as before this PR? I ask because a `list` may be slower, as we iterate over `prospectives` a bit after:

```python
        # Internal deps
        for ld in prospectives:
            dirseg = os.path.join(self.environment.get_build_dir(), self.get_target_dir(ld))
            result.add(dirseg)
```

Thanks a lot for reviewing the PR! :)

Reply:
No, I mean a `list`; otherwise you would not preserve the order as you wanted. Inserting into a list is faster than inserting into a set, because it doesn't need to test whether the element is already present or not. But I may be wrong on the overall timing; this is why I said you should really profile the code to verify which one is faster.

Reply:
Well, actually it ain't much work... admittedly, I don't know which version is faster 🙂

Reply:
You can use the `--profile-self` to generate profile information. It use the standard profile module from Python. To visualize the results, I like to use `snakeviz` (https://jiffyclub.github.io/snakeviz/)

Reply:
so `OrderedSet` is clearly faster in this case. I'm surprised, but it is possible...